tw/debug/UnlockOptionStateReducer.js

```diff
@@ -0,0 +1,50 @@
+; /**** game/Modules/Scavenging/Widgets/UnlockOptionDialog/UnlockOptionStateReducer.js ****/
+define("Ig/TribalWars/Modules/Scavenging/Widgets/UnlockOptionDialog/UnlockOptionStateReducer", ["Ig/TribalWars/Modules/Common/Widget/AbstractStateReducer", "Ig/TribalWars/Modules/Scavenging/Widgets/UnlockOptionDialog/StateReductionActions"], function(e, t) {
+    "use strict";
+
+    function o() {
+        e.apply(this)
+    }
+    return o.prototype = Object.create(e.prototype), $.extend(o.prototype, {
+        newStateFromNothing: function() {
+            return {
+                title: null,
+                res_label: _("5e2bc1b784edf7c3144e867ecacdfb38"),
+                wood: null,
+                stone: null,
+                iron: null,
+                wood_affordable: null,
+                stone_affordable: null,
+                iron_affordable: null,
+                duration_seconds: null,
+                button_text: _("7e7123b0d269ad5d9ec8d12c52a8ed8a"),
+                is_button_disabled: null,
+                disabled_explanation: "",
+                waiting_unlock_request: !1,
+                resource_forecast_html: ""
+            }
+        },
+        newStateFromModels: function(e, t, o) {
+            var n = t.village,
+                a = t.option,
+                r = $.extend(!0, {}, e);
+            null === r.title && (r.title = s(_("67650110ddd6cbe7438c9b96d7e4994b"), a.getName()));
+            var l = a.getUnlockCost();
+            null === r.wood && (r.wood = l.wood, r.stone = l.stone, r.iron = l.iron, r.duration_seconds = a.getUnlockDurationSeconds()), r.wood_affordable = n.res.wood >= r.wood, r.stone_affordable = n.res.stone >= r.stone, r.iron_affordable = n.res.iron >= r.iron;
+            var i = ResourcesForecaster.getForecast(l, game_data.village, t.rate_schedule, t.amount_schedule),
```

Comment:
referencing the wrong village (game_data.village).

game_data.village is the currently active village. This needs to reference a Village object representing the village which the popup is supposed to be for.

Pass it in as a model to UnlockOptionWidget. This will take a bit more work.